cpp/src/arrow/gpu/CMakeLists.txt

```diff
@@ -43,7 +43,7 @@ ADD_ARROW_LIB(arrow_gpu
   DEPENDENCIES metadata_fbs
   SHARED_LINK_FLAGS ""
   SHARED_LINK_LIBS ${ARROW_GPU_SHARED_LINK_LIBS}
-  STATIC_LINK_LIBS ""
+  STATIC_LINK_LIBS ${ARROW_GPU_SHARED_LINK_LIBS}
```

Comment:
This is probably ok but looks weird that the `STATIC_LINK_LIBS` list `…_SHARED_LINK_LIBS`. Can you add a comment that this is intended?

Reply:
Done.